Clarify definition of off-path (#4462)
quicwg/base-drafts: draft-ietf-quic-transport.md

relies on reception of ICMP messages (e.g., IPv6 Packet Too Big messages) that
indicate when an IP packet is dropped because it is larger than the local router
MTU. DPLPMTUD can also optionally use these messages.  This use of ICMP messages

> is potentially vulnerable to attacks by elements that cannot observe packets

Reviewer: elements or entities or does it matter? "path elements" is used only once elsewhere in the I-D.

> is potentially vulnerable to attacks by elements that cannot observe packets,

Reviewer 2: ```suggestion
is potentially vulnerable to attacks by elements that cannot observe packets
```

Connection ID MUST be copied from the Destination Connection ID of the received
packet, which is initially randomly selected by a client.  Echoing both
connection IDs gives clients some assurance that the server received the packet
and that the Version Negotiation packet was not generated by an entity that

> did not receive the Initial packet.

Reviewer 3: ```suggestion
did not observe the Initial packet.
```

In order to prevent tampering by version-unaware middleboxes, Initial packets
are protected with connection- and version-specific keys (Initial keys) as
described in {{QUIC-TLS}}.  This protection does not provide confidentiality or
integrity against attackers that can observe packets, but provides some level of

> protection against attackers that cannot also observes packets.

Reviewer 4: ```suggestion
protection against attackers that cannot observe packets.
```

are protected with connection- and version-specific keys (Initial keys) as
described in {{QUIC-TLS}}.  This protection does not provide confidentiality or

> integrity against on-path attackers, but provides some level of protection

Reviewer 3: ```suggestion
integrity against attackers that can observe the packets, but provides some level of protection
```

> against attackers that cannot also receive packets.

Reviewer 3: ```suggestion
against attackers that cannot also observes packets.
```

An off-path attacker cannot:

> - Modify packets sent by endpoints

Reviewer 5: Ah, good call

An off-path attacker cannot:

- Modify packets sent by endpoints
- Delay packets
- Drop packets
- Reorder original packets

An off-path attacker can create modified copies of packets that it has observed
and inject those copies into the network, potentially with spoofed source and
destination addresses.

For the purposes of this discussion, it is assumed that an off-path attacker

> has the ability to observe and re-inject a modified copy of a packet into the

Reviewer 4: ```suggestion
has the ability to inject a modified copy of a packet into the
```
Given that we've just talked about the observe-modify-reinject process in the previous sentence, I think we're safe to shorthand this to injecting a modified packet.
